### ISSUE TITLE
Add isLoadable method to RepositoryLocation

### ISSUE THREAD

**Issue by madoar** [PlayOnLinux/POL-POM-5#905]
With regards to #901 and similar issues, which occur when a repository couldn't be read/loaded, we should add an indication about the readability/loadability of a repository to the `RepositoryLocation`.
This method should not be persisted by jackson.

The result of this method can then be used to visually show the user in the settings tab if a repository can't be opened.
We could for example show the background of the "bad" repository location in a red color, or display an attention symbol next to the repository location.
What do you think?

**Comment by plata** [comment]
Good idea. It's easy to have a typo in your repository URL and with this solution, you should immediately see it. Maybe we should also add a sanity check in the "add repository" dialog (e.g. `if localRepositoryFile.exists()`).

**Comment by madoar** [comment]
What do you think should the `isLoadable` method check for the different repository types?

**Comment by plata** [comment]
classpath: The given classpath exists.
local: The given folder exists.
git: `git ls-remote` succeeds.

**Comment by madoar** [comment]
Hmm:

- classpath: I'm not sure this is possible, after a short search on google I think it's only possible to check if a given class exists and not a package, for which we are searching.
- local: This should be easy
- git: Do you know how the result should look like for a nonexistent repository? If we use `JGit`, I think we may need to check for an exception, but I'm not sure... Take a look [here](https://github.com/centic9/jgit-cookbook/blob/master/src/main/java/org/dstadler/jgit/porcelain/ListRemotes.java) and [here](https://github.com/centic9/jgit-cookbook/blob/master/src/main/java/org/dstadler/jgit/porcelain/ListRemoteRepository.java)

**Comment by plata** [comment]
- classpath: How about `getClass().getClassLoader().getResource("javax/servlet");`? Or would it be possible to restrict the selection
- git: Never did this with JGit. In plain git, you get `fatal: '<URL>' does not appear to be a git repository`. However, it should be easy to figure out if JGit gives an exception or just an empty return value.

**Comment by madoar** [comment]
- classpath: I don't think this works, because we don't really wantto test if `javax.servlet` exists, but if a file inside our repository exists. To do this, we would need to have a file, that is contained in all Phoenicis repositories. I believe such a file would be helpful in any case, because it would help us verify that the folder/git repository really is a Phoenicis repository.
- git: You're right it should be easy to test, what feedback JGit provides for an non existent repository

**Comment by plata** [comment]
Shouldn't the resource check ensure that the given directory exists in the classpath? It might be empty but that could happen anyways (e.g. also for a local folder).
